Lib/glyphsLib/builder/anchors.py

```diff
@@ -95,18 +113,35 @@ def _get_anchor_data(anchor_data, ufo, components, anchor_name):
         anchor_data[anchor.name] = t.transformPoint((anchor.x, anchor.y))
 
 
-def _adjust_anchors(anchor_data, ufo, component):
-    """Adjust anchors to which a mark component may have been attached."""
+def _anchor(_glyph, _component):
+    """Pull component’s named anchor from Glyph.lib"""
+    if COMPONENT_INFO_KEY in _glyph.lib:
+        for _anchorLib in _glyph.lib[COMPONENT_INFO_KEY]:
+            if (
+                "anchor" in _anchorLib
+                and "name" in _anchorLib
```

Comment:
More than one component with same name is possible, but in our case of a ligature glyph we want the component that’s attached to a certain named anchor (`top_2`), so I'm checking for that and believe that the component index is irrelevant here. I have nothing to compare that index against.

Reply:
The "name" in `_anchorLib` is the name of the component not the name of the anchor (that's the `_anchorLib["anchor"]`); there can be more than one components with the same name. The component index should be enumerated when we do `for component in parent.components` and passed down to the `_anchor` function.
The COMPONENT_INFO_KEY is the additional Glyphs-specific metadata that is associated with a particular component, identified by its index, which is unique, not the name. So you should match them by index. @madig am I correct?

Reply:
I suggest you add a test with a composite glyph with more than one component with the same name, where each has different named anchors; your current code will only match and use the component info of the first component with a given name.

Reply:
Okay, I got what you mean now. I'm on it.

Reply:
Actually this messes up even in Glyphs using `GSLayer.anchorsTraversingComponents()`. Instead of `top_1` and `top_2` both moving upwards, both stay where they were, and a previously non-existent `top` anchor appears where I though `top_1` would end up. Nothing such happens on the `top_2` side, it just stays in its old place :/

I can still implement this if you like, but I must add that this will be ahead of Glyphs and therefore inconsistent ;)

Reply:
that looks like a bug which Georg can easily fix, I prefer that we do the right thing.

Reply:
Okay, I'll let him know. 
But I still don't get your idea with the index, sorry.
This is the lib regarding the anchors of the glyph with the double components: `[{'name': 'hamzaabove-ar', 'index': 1, 'anchor': 'top_2'}, {'name': 'hamzaabove-ar', 'index': 2, 'anchor': 'top_1'}]`
What do you want me to compare the index against? I have only the anchor names in `anchor_data` and compare them against what is the result of `_anchor()` to identify when to apply transformations.

Reply:
against the index of the component in the list of components in the parent composite glyph. We need to record that alongside the component name to fully indentify the specific component info metadata (where the anchor name is) that is stored in the glyph.lib (a component doesn't have a lib of its own).
The index of the component is its position in the list of glyph.components. When you iterate over the components, you use `for i, component in enumerate(glyph.components):` and store that index and pass it down to where you need it

Reply:
The test is a bit more tricky. The test glyph I created was fictional. A real-world `lamAlef-ar` with two components however propagates the anchors correctly in Glyphs, such as `lamVabove_alefWasla-ar`. So I need to find a test glyphs that's defined in the `GlyphData.xml` that has two identical components.

Reply:
I just tried and the "top_2" vs "top" was correct when running `GSLayer.anchorsTraversingComponents()`. Can you point me to a sample file where it doesn't work?

Reply:
I did run into the issue you pointed out with my fictional test glyph with two duplicate marks, and solved it.